src/clinical/api/clinical-api.ts

```diff
@@ -186,15 +183,39 @@ class ClinicalController {
 
 		const donorEntityData = await service.getDonorEntityData(donorIds);
 
+		const entityRecords = donorEntityData[0].records;
+
+		// File Table can request multiple programs
+		const donorPrograms = entityRecords
+			.map((record) => record.program_id as string)
+			.filter((programId, index, array) => programId && array.indexOf(programId) === index);
+
 		const date = currentDateFormatted();
 		const fileName = `filename=Donor_Clinical_Data_${date}.zip`;
+
+		const exceptions = await Promise.all(
+			donorPrograms.map(async (programShortName) => {
+				const programDonors = entityRecords
+					.filter((record) => record.donor_id && record.program_id === programShortName)
+					.map((record) => record.donor_Id as number);
```

Comment:
Can you do `.map((record) => Number(record.donor_Id))` instead of the type assertion? I'm not sure what type these are coming back as and this will ensure the type is converted correctly. It will also take weird strings and make them into `NaN` so you may want a filter step after this conversion. 

Also, should I be concerned about the mixed case of `donor_id` and `donor_Id` on neighbouring lines?

Reply:
Added numeric conversion, fixed property name